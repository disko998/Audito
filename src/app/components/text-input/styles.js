import { makeStyles } from '@material-ui/core/styles'

export const useStyles = makeStyles(() => ({
  textField: {
    width: '100%',
    marginTop: 10,
    marginBottom: 10,
  },
  label: error => ({
    color: error ? '#f44336' : '',
    textTransform: 'capitalize',
  }),
}))
