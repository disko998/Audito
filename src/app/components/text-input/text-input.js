import React from 'react'
import {
    Input,
    InputAdornment,
    IconButton,
    FormControl,
    InputLabel,
} from '@material-ui/core'
import Visibility from '@material-ui/icons/Visibility'
import VisibilityOff from '@material-ui/icons/VisibilityOff'

import { useStyles } from './styles'

export const TextInput = ({ label, secure, onIconPress, error, ...otherProps }) => {
    const classes = useStyles(error)

    return (
        <FormControl className={classes.textField}>
            <InputLabel className={classes.label}>{label}</InputLabel>
            <Input
                fullWidth
                type={secure ? 'password' : 'text'}
                endAdornment={
                    onIconPress && (
                        <InputAdornment position='end'>
                            <IconButton
                                aria-label='toggle password visibility'
                                onClick={onIconPress}
                            >
                                {secure ? <Visibility /> : <VisibilityOff />}
                            </IconButton>
                        </InputAdornment>
                    )
                }
                error={error}
                {...otherProps}
            />
        </FormControl>
    )
}

TextInput.defaultProps = {
    label: 'Type something...',
    secure: false,
    onIconPress: false,
    error: false,
}
