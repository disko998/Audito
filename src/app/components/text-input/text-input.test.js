//@ts-nocheck
import React from 'react'
import { shallow } from 'enzyme'
import { TextInput } from './text-input'

it('text input render', () => {
    expect(shallow(<TextInput />)).toMatchSnapshot()
})
