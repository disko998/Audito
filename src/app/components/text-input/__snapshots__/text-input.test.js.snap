// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text input render 1`] = `ShallowWrapper {}`;
